Ported AI Navmap

ScarKy0 opened the pull request:
<!-- Guidelines: https://docs.spacestation14.io/en/getting-started/pr-guideline -->

## About the PR
<!-- What did you change? -->
Ported the AI Navmap view from https://github.com/carpmosia/carpmosia/pull/80 with permission from the author.

## Why / Balance
<!-- Discuss how this would affect game balance or explain why it was changed. Link any relevant discussions or issues. -->
One of the biggest wanted changed regarding AI both from players and the silicon workgroup.
Also required for several other changes we want to make later down the line, such as making cameras not grant vision without power, but those would require some additional tweaks.

## Technical details
<!-- Summary of code changes for easier review. -->
Navmap rendering code has been added to the StationAiOverlay and drawing the static has been removed.

## Media
<!-- Attach media if the PR makes in-game changes (clothing, items, features, etc).
Small fixes/refactors are exempt. Media may be used in SS14 progress reports with credit. -->

https://github.com/user-attachments/assets/c7ad82d5-cd2a-4e12-983b-228d5e149714

## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [X] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [X] I have added media to this PR or it does not require an in-game showcase.
<!-- You should understand that not following the above may get your PR closed at maintainer’s discretion -->

## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them.
This will be posted in #codebase-changes. -->

**Changelog**
<!-- Add a Changelog entry to make players aware of new features or changes that could affect gameplay.
Make sure to read the guidelines and take this Changelog template out of the comment block in order for it to show up.
Changelog must have a :cl: symbol, so the bot recognizes the changes and adds them to the game's changelog. -->
:cl:
- add: AI now uses Navmap vision instead of the static shader.
<!--
:cl:
- add: Added fun!
- remove: Removed fun!
- tweak: Changed fun!
- fix: Fixed fun!
-->

ScarKy0 committed:
init

ScarKy0 commented:
I knew there's a catch to this somewhere :godo:

SnappingOpossum commented:
Keep in mind [carpmosia#103](https://github.com/carpmosia/carpmosia/issues/103) as well.

ScarKy0 commented:
I'll look into it, though this is still code I have basically no experience in

ArtisticRoomba committed:
memory is not free are you guys crazy??????
- kill ValueList and just reuse lists
- stop making copies of a massive list to pass in
- other misc evil hidden allocations here and there

todo:
- port actual sane code
- attempt to fix occlusion issue by bumping lines (will probably run into issues going around corners)
- make it look better smile

ScarKy0 commented:
Closing it because its stale and navmap view needs to be handled in a different way.
Instead of kidnapping navmap code this should be it's own renderer, or else we end up with hardcoding AI things into the navmap system. Either that or navmap needs to be made more robust to allow for appending onto overlays by itself.
